core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlRelationBuilder.scala

```diff
@@ -66,13 +65,14 @@ class TSqlRelationBuilder(override val vc: TSqlVisitorCoordinator)
     case None =>
       ctx match {
         case qs if qs.querySpecification() != null => qs.querySpecification().accept(this) // TODO: Implement set ops
-        case _ =>
-          // TODO: Implement this style of UNION
-          ir.UnresolvedRelation(
-            ruleText = contextText(ctx),
-            message = s"This style of UNION specification is as yet supported",
-            ruleName = vc.ruleName(ctx),
-            tokenName = Some(tokenName(ctx.getStart)))
+        case qe if qe.queryExpression() != null =>
+          val lhs = qe.queryExpression(0).accept(this)
+          Option(qe.queryExpression(1)).map(_.accept(this)) match {
```

Comment:
```suggestion
          qe.queryExpression().asScala.lift(1).map(_.accept(this)) match {
```
avoids `IndexOutOfBoundException` if there is only one `queryExpression`

Reply:
Thanks for this tip: it triggered me to realise I misunderstood something, and now I have a much cleaner version.